tests/phpunit/tests/user/queryCache.php

```diff
@@ -0,0 +1,706 @@
+			'count_total' => false,
+		);
+
+		$query2        = new WP_User_Query( $args );
+		$users2        = wp_list_pluck( $query2->get_results(), 'ID' );
+		$users_total2  = $query2->get_total();
+		$queries_after = get_num_queries();
+
+		$this->assertNotSame( $queries_before, $queries_after, 'Assert that queries are run' );
+		$this->assertNotSame( $users_total1, $users_total2, 'Assert that totals do not match' );
+		$this->assertSameSets( $users1, $users2, 'Asset that results of query match' );
+	}
+
+	/**
+	 * @ticket 40613
+	 */
+	public function test_query_cache_results() {
+		$args = array(
+			'cache_results' => true,
+		);
+
+		$query1 = new WP_User_Query( $args );
+		$users1 = wp_list_pluck( $query1->get_results(), 'ID' );
+
+		$queries_before = get_num_queries();
+
+		$args = array(
+			'cache_results' => false,
+		);
+
+		$query2        = new WP_User_Query( $args );
+		$users2        = wp_list_pluck( $query2->get_results(), 'ID' );
+		$queries_after = get_num_queries();
+
+		$this->assertNotSame( $queries_before, $queries_after, 'Assert that queries are run' );
+		$this->assertSameSets( $users1, $users2, 'Asset that results of query match' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @expectedDeprecated WP_User_Query
+	 */
+	public function test_query_cache_who() {
+		$args = array(
+			'who'    => 'authors',
+			'fields' => array( 'ID' ),
+		);
+
+		$query1       = new WP_User_Query( $args );
+		$users1       = $query1->get_results();
+		$users_total1 = $query1->get_total();
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_User_Query( $args );
+		$users2         = $query2->get_results();
+		$users_total2   = $query2->get_total();
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( $queries_before, $queries_after, 'Assert that no queries are run' );
+		$this->assertSame( $users_total1, $users_total2, 'Assert that totals do match' );
+		$this->assertSameSets( $users1, $users2, 'Asset that results of query match' );
+	}
+
+	/**
+	 * @dataProvider data_returning_field_subset_as_array
+	 * @ticket 40613
+	 * @param array $args
+	 */
+	public function test_query_cache( $args ) {
+		$query1       = new WP_User_Query( $args );
+		$users1       = $query1->get_results();
+		$users_total1 = $query1->get_total();
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_User_Query( $args );
+		$users2         = $query2->get_results();
+		$users_total2   = $query2->get_total();
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( 0, $queries_after - $queries_before, 'Assert that no queries are run' );
+		$this->assertSame( $users_total1, $users_total2, 'Assert that totals do match' );
+		$this->assertSameSets( $users1, $users2, 'Asset that results of query match' );
+	}
+
+	/**
+	 * Data provider
+	 *
+	 * @return array
+	 */
+	public function data_returning_field_subset_as_array() {
+		$data = array(
+			'id'                    => array(
+				'args' => array( 'fields' => array( 'id' ) ),
+
```

Comment:
```suggestion
				'args' => array( 'fields' => array( 'id' ) ),
```

I think when there is only one parameter it's fine to bypass the name in the dataprovider -- i will check with others.